engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/source/KafkaAvroSourceFactory.scala

```diff
@@ -21,20 +22,29 @@ class KafkaAvroSourceFactory[T:TypeInformation](val schemaRegistryProvider: Sche
 
   override def contextTransformation(context: ValidationContext, dependencies: List[NodeDependencyValue])
                                     (implicit nodeId: ProcessCompilationError.NodeId): NodeTransformationDefinition = {
-    case TransformationStep(Nil, _) => NextParameters(initialParameters)
-    case TransformationStep((KafkaAvroFactory.TopicParamName, DefinedEagerParameter(topic:String, _)) :: Nil, _) =>
-       NextParameters(List(versionParam(topic)), Nil, None)
-    case TransformationStep((KafkaAvroFactory.TopicParamName, DefinedEagerParameter(topic:String, _)) ::
-      (KafkaAvroFactory.SchemaVersionParamName, DefinedEagerParameter(version:Integer, _)) ::Nil, _) =>
-      val result = createSchemaRegistryProvider(topic, version).typeDefinition
+    case TransformationStep(Nil, _) =>
```

Comment:
Does this code is duplicated at `KafkaAvroSinkFactory`? Maybe we can extract it to some Mixin..?

Reply:
I extracted a bit